qcodes/dataset/experiment_container.py

```diff
@@ -180,18 +180,23 @@ def __repr__(self) -> str:
 
 # public api
 
-def experiments()->List[Experiment]:
+def experiments(conn: Optional[ConnectionPlus] = None) -> List[Experiment]:
     """
     List all the experiments in the container (database file from config)
 
+    Args:
+        conn: connection to the database. If not supplied, a new connection
+          to the DB file specified in the config is made
+
     Returns:
         All the experiments in the container
     """
-    log.info("loading experiments from {}".format(get_DB_location()))
-    rows = get_experiments(connect(get_DB_location(), get_DB_debug()))
+    log.info("loading experiments from {}".format(conn))
```

Comment:
to my knowledge printing the connection object is quite useless. However, `ConnectionPlus` object has `.path_to_dbfile` attribute which can be used for this log message after the next line is executed. (i mean to suggest to move this logging line after the next line)